src/main/kotlin/me/ebonjaeger/perworldinventory/conversion/ConvertExecutor.kt

```diff
@@ -0,0 +1,151 @@
+package me.ebonjaeger.perworldinventory.conversion
+
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import com.onarandombox.multiverseinventories.ProfileTypes
+import com.onarandombox.multiverseinventories.api.profile.PlayerProfile
+import com.onarandombox.multiverseinventories.api.profile.WorldGroupProfile
+import com.onarandombox.multiverseinventories.api.share.Sharables
+import me.ebonjaeger.perworldinventory.ConsoleLogger
+import me.ebonjaeger.perworldinventory.Group
+import me.ebonjaeger.perworldinventory.GroupManager
+import me.ebonjaeger.perworldinventory.initialization.DataDirectory
+import me.ebonjaeger.perworldinventory.serialization.InventorySerializer
+import me.ebonjaeger.perworldinventory.serialization.PotionSerializer
+import org.bukkit.GameMode
+import org.bukkit.OfflinePlayer
+import org.bukkit.potion.PotionEffect
+import java.io.File
+import java.io.FileWriter
+import java.nio.file.Files
+import java.util.*
+import javax.inject.Inject
+
+/**
+ * Class that performs converting operations.
+ */
+class ConvertExecutor @Inject constructor(private val groupManager: GroupManager,
+                                          @DataDirectory private val dataDirectory: File)
+{
+
+    var mvGroups: List<WorldGroupProfile>? = null
```

Comment:
Am I being stupid or does this never get set to something?
Oh, it gets set in the ConvertService. Maybe there's a better way to handle this. I'm a little too short on time now to make a suggestion (I want to go to bed in 5 :smile:)